Lib/datetime.py

```diff
@@ -1572,6 +1572,17 @@ def _fromtimestamp(cls, t, utc, tz):
             # 23 hours at 1969-09-30 13:00:00 in Kwajalein.
             # Let's probe 24 hours in the past to detect a transition:
             max_fold_seconds = 24 * 3600
+
+            # On Windows localtime_s throws an OSError for negative values,
+            # thus we can't perform fold detection for values of time less
+            # than the max time fold. See comments in _datetimemodule's
+            # version of this method for more details.
+            try:
+                converter(-1)
```

Comment:
Is this a way to detect windows or a windows-like system? Any such detection should be done in the module scope to avoid run-time penalty.  Why not just check os.platform()?

Reply:
It looks like the test suite uses `sys.platform.startswith('win')`, would that be acceptable? My reasoning for making it a generic detection was just in case any other platforms have the same problem, though that might be over-engineering it.

Reply:
But this doesn't reflect the C implementation, which *is* only checking if it's being run on Windows.

Reply:
Fair enough, this is now a platform check. There is prior art in another stdlib module to suggest that the check from above is fine: https://github.com/python/cpython/blob/91106cd9ff2f321c0f60fbaa09fd46c80aa5c266/Lib/_bootlocale.py#L10